will/backends/io_adapters/slack.py

```diff
@@ -356,6 +366,34 @@ def send_message(self, event):
         )
         self.handle_request(r, data)
 
+    def upload_file(self, event):
+        data = {}
+        if hasattr(event, "kwargs"):
+            data.update(event.kwargs)
+
+        for attribute in event.kwargs:
+            if attribute in ("title", "filetype", "filename", "initial_comment", "is_private_chat"):
+                data.update({
+                    attribute: event.kwargs[attribute]
+                })
+
+        if isinstance(event.content, bytes):
+            data.update({
+                "file": event.content,
+            })
+        else:
+            data.update({
+                "content": event.content,
+            })
+
+        data = self.set_data_channel_and_thread(event, data=data)
+
+        ret = self.client.api_call("files.upload",
```

Comment:
I gave this a try and it seems to get stuck here. The api call never returns.

Reply:
Thanks  for testing it, I'll walk through the code and see what's happening. At this point it's passing everything to the slack API so it may have a fault in how the response code it checked.